packages/astro/src/core/render/result.ts

```diff
@@ -194,13 +194,23 @@ export function createResult(args: CreateResultArgs): SSRResult {
 					result.cookies = cookies;
 					return cookies;
 				},
+				get preferredLocale(): string | undefined {
+					if (locales) {
+						return computePreferredLocale(request, locales);
```

Comment:
It would be good to cache the result of the compute, since it's relatively expensive, so the cost only happens on the first time. If you look at how we do this in cookies, we use a closure variable and check if it is already defined inside of the get().

Same for the other places where this compute occurs.

Reply:
Done here https://github.com/withastro/astro/pull/8997/commits/2060d027aaff5d555f183c3c8225e08ed1b5af2f